workers/unity/Packages/com.improbable.gdk.core/Components/NewlyAddedSpatialOSEntity.cs

```diff
@@ -7,8 +7,11 @@ namespace Improbable.Gdk.Core
     ///     This component is automatically added to an entity upon its creation and automatically removed at the end of the
     ///     same frame.
     /// </summary>
-    [RemoveAtEndOfTick]
-    public struct NewlyAddedSpatialOSEntity : IComponentData
+    public struct NewlyAddedSpatialOSEntity : IComponentData, RemoveAtEndOfTick
     {
+        public void RemoveComponent(EntityCommandBuffer commands, Entity entity)
+        {
+            commands.RemoveComponent<NewlyAddedSpatialOSEntity>(entity);
+        }
```

Comment:
This doesn't look very nice, and the implementation is effectively boilerplate with component type inserted.
Unfortunately, as EntityCommandBuffer RemoveComponent method is generic, we have to explicitly declare it somewhere for the AOT compiler.
This approach allows us to keep compile-time safety, where compiler will complain if RemoveComponent isn't implemented
I am open to other suggestions!

Reply:
Agreed. It doesn't make me feel warm an fuzzy but it's the best solution I can think of that allows us to keep the feature. The question then is how much do we get from this feature and is the boiletplate too annoying. I think this might be fine.

Annoying C# doesn't let us inherit functionality in structs or we could have made the interface generic.

Reply:
Mentioned further down - I'd forgotten that this will create boxing allocations on each access, which might sink this idea.

Reply:
Actually I'm wrong we can still use this but it will require some funky reflection code. If we make a static function then we can find that via reflection when we find the types in the cleanup system. We can store each of those in an action and call them at runtime. Not exactly pretty but it would work. That doesn't allocate on removal but it does place a larger burden on creating temporary components (although you can give good error messages as we can diagnose the missing function perfectly).

Reply:
Another possibility is users registering an action and type in the clean system at startup.

Reply:
Another option is we don't inject anything and we only use injections to get a list of entity then use the manger directly.

Reply:
Wasn't one of the points of this to remove the need for reflection? :)

Reply:
Yeah, this is no longer needed, I'll remove it